REPOSITORY_STANDARDS.md

```diff
@@ -0,0 +1,89 @@
+# Repository standards
+
+Here is a list of repository standards we have for our repositories. Much of the below is automated by our `dp-cli` tool for creating repositories but is documented here for completeness.
```

Comment:
Consider adding a link for the `dp-cli` reference as QoL consideration.

We should also update the `dp-cli` documentation to remind people that changes there need to be aligned to this standard.